physionet-django/project/management/commands/get_metrics_data.py

```diff
@@ -0,0 +1,267 @@
+"""
+Command to:
+- Parse NGINX log files and write their metrics data to the database
+"""
+
+import datetime
+
+from django.core.management.base import BaseCommand
+from django.core.exceptions import ObjectDoesNotExist
+from django.utils import timezone
+
+from project.models import CoreProject, PublishedProject, Metrics
+
+
+class Command(BaseCommand):
+    """
+    Command to parse a log file and load data to the Metrics table.
+    """
+    help = "Parses a log file and loads data to the Metrics table."
+
+    def add_arguments(self, parser):
+        parser.add_argument('files', nargs='+', type=str,
+                            help='Specify log file(s) to parse')
+        parser.add_argument('-c', '--is_cron', type=str_to_bool, default=False,
+                            help='If True, run as cron job')
+        parser.add_argument('-n', '--now', type=str, default=None,
+                            help='Use given date as "now" (format DD/MM/YYYY)')
+
+    def handle(self, *args, **options):
+        """Retrieves metrics data from the latest log file, or as specified.
+
+        Retrieves view count from most recent log file, or from the log file(s)
+            specified in **options.
+
+        Args:
+            **options:
+                files [filename(s)]: One or more NGINX log file(s) to parse.
+                -c [bool]: If True, runs update from log as cron job.
+                -n [date]: If running a cron job, treats the given date as
+                    "now". Format: DD/MM/YYYY.
+
+        Returns:
+            None
+
+        Raises:
+            InvalidLogError: An error occurred obtaining metrics data from the
+                log file.
+            DateError: The date in the log file does not match the expected
+                date (if run as a cron job, the log file date should be the day
+                before).
+        """
+        files = options['files']
+        is_cron = options['is_cron']
+        now = str_to_datetime(options['now'])
+
+        for filename in files:
+            try:
+                validate_log_file(filename, is_cron, now)
+                update_metrics(filename)
+            except InvalidLogError:
+                raise InvalidLogError(
+                    f'Log file {filename} is not a valid NGINX log')
+            except DateError:
+                raise DateError('Log file has incorrect date')
+
+
+def log_parser(filename, data=None):
```

Comment:
Please could you how you anticipate the `data` argument being used, and perhaps clarify this in the docstring? I don't see it being used in this pull request, so is it needed?

Reply:
You're right, it's not really used anywhere--I took out the `data` argument.